hasher-matcher-actioner/hmalib/lambdas/api/submit.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import bottle
+import boto3
+import base64
+
+from dataclasses import dataclass, asdict
+from mypy_boto3_dynamodb.service_resource import Table
+import typing as t
+
+from hmalib.lambdas.api.middleware import jsoninator, JSONifiable, DictParseable
+from hmalib.common.logging import get_logger
+
+logger = get_logger(__name__)
+s3_client = boto3.client("s3")
+dynamodb = boto3.resource("dynamodb")
+
+
+@dataclass
+class SubmitContentRequestBody(DictParseable):
+    submission_type: str  # TODO Enum
+    content_id: str
+    content_type: str  # TODO Enum
+    content_ref: t.Union[str, bytes]
```

Comment:
I think another kind of submission might take a URL to the content. If we did that, maybe we could use content_ref for the URL and a submission type of, say, URL? More of a conversation / idea sharing comment.

Reply:
Yes, that is the plan!